### ISSUE TITLE
Using tagNamePreference breaks other rules

### ISSUE THREAD

**Issue by edg2s** [gajus/eslint-plugin-jsdoc#1559]
## Expected behavior
`jsdoc/require-yields-type` reports errors even if I have set my preferred tag name to `yield` via `tagNamePreference`

## Actual behavior
The error is not detected

## ESLint Config
```json
{
	"extends": [ "plugin:jsdoc/recommended" ],
	"settings": {
		"jsdoc": {
			"tagNamePreference": {
				"returns": "return",
				"yields": "yield"
			}
		}
	},
	"rules": {
		"jsdoc/require-yields-type": "error"
	}
}
```
## ESLint sample

```js
/**
 * @yield
 */
```
